WireConversations/Sources/WireConversationsUI/WireConversationTypePicker.swift

```diff
@@ -0,0 +1,162 @@
+                self = .channel
+            case .group:
+                self = .group
+            }
+        }
+    }
+
+    @State private var selection: WireConversationType?
+
+    private let displayedItems: [DisplayedItem]
+    private let onConversationTypeSelected: @Sendable (WireConversationType) -> Void
+
+    package init(
+        availableConversationTypes: Set<WireConversationType>,
+        onConversationTypeSelected: @escaping @Sendable (WireConversationType) -> Void
+    ) {
+        let orderedConversationTypes: [WireConversationType] = [.channel, .group]
+
+        self.displayedItems = Array(
+            orderedConversationTypes
+                .filter(availableConversationTypes.contains)
+                .map { [DisplayedItem($0)] }
+                .joined(separator: [.divider])
+        )
+
+        self.onConversationTypeSelected = onConversationTypeSelected
+    }
+
+    package var body: some View {
+        VStack(spacing: 0) {
+            ForEach(displayedItems, id: \.hashValue) { conversationType in
+                view(for: conversationType)
+            }
+        }
+        .background(.white)
+        .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
+        .onChange(of: selection) { newValue in
+            guard let newValue else { return }
+            onConversationTypeSelected(newValue)
+        }
+    }
+
+    @ViewBuilder
+    private func view(for item: DisplayedItem) -> some View {
+        switch item {
+        case .channel:
+            channelItem()
+                .padding(.horizontal, 16)
+                .frame(minHeight: Constants.minRowHeight)
+        case .divider:
+            Rectangle()
+                .fill(ColorTheme.Backgrounds.background.color)
+                .frame(height: 1)
+                .padding(.leading, 56)
+        case .group:
+            groupItem()
+                .padding(.horizontal, 16)
+                .frame(minHeight: Constants.minRowHeight)
+        }
+    }
+
+    func channelItem() -> some View {
+        Button(action: {
+            selection = .channel
+            onConversationTypeSelected(.channel)
+        }, label: {
+            HStack(alignment: .center, spacing: Constants.verticalSpacing) {
+                iconView(for: "wire_conversation_channel_icon")
+                VStack(alignment: .leading) {
+                    Text(L10n.Localizable.Conversation.Create.Channel.title)
+                        .wireTextStyle(.body2)
+                        .foregroundStyle(ColorTheme.Backgrounds.onSurface.color)
+                    Text(L10n.Localizable.Conversation.Create.Channel.subtitle)
+                        .wireTextStyle(.h4)
+                        .foregroundStyle(ColorTheme.Base.secondaryText.color)
+                }
+                Spacer()
+                chevronView()
+            }
+        })
+    }
+
+    func groupItem() -> some View {
+        Button(action: {
+            selection = .group
```

Comment:
isn't onConversationTypeSelected missing?
```suggestion
            selection = .group
            onConversationTypeSelected(.group)
```

Reply:
That’s what the onChange is for, which incidentally enables deduplication of events. I’ll fix the other method to remove the call to the closure.